tracer/src/Datadog.Trace/Configuration/ExporterSettings.cs

```diff
@@ -177,142 +199,180 @@ public int PartialFlushMinSpans
+            if (!string.IsNullOrWhiteSpace(tracesPipeName))
+            {
+                TracesTransport = TracesTransportType.WindowsNamedPipe;
+                TracesPipeName = tracesPipeName;
+                SetAgentUri(agentHost ?? DefaultAgentHost, agentPort ?? DefaultAgentPort); // this one can throw
+                return;
+            }
 
-            if (hasExplicitHostOrPortSettings)
+            // This property shouldn't have been introduced. We need to remove it as part of 3.0
+            // But while it's here, we need to handle it properly
+            if (!string.IsNullOrWhiteSpace(tracesUnixDomainSocketPath))
             {
-                if (agentHost?.StartsWith(UnixDomainSocketPrefix) ?? false)
+                if (TrySetAgentUriAndTransport(UnixDomainSocketPrefix + tracesUnixDomainSocketPath))
                 {
-                    traceTransport = TracesTransportType.UnixDomainSocket;
-                    TracesUnixDomainSocketPath = agentHost;
+                    return;
                 }
-                else
+            }
+
+            if ((agentPort != null && agentPort != 0) || agentHost != null)
+            {
+                // Agent port is set to zero in places like AAS where it's needed to prevent port conflict
+                // The agent will fail to start if it can not bind a port, so we need to override 8126 to prevent port conflict
+                // Port 0 means it will pick some random available port
+
+                if (TrySetAgentUriAndTransport(agentHost ?? DefaultAgentHost, agentPort ?? DefaultAgentPort))
                 {
-                    // The agent host is explicitly configured, we should assume UDP for metrics
-                    forceMetricsOverUdp = true;
-                    traceTransport = TracesTransportType.Default;
+                    return;
                 }
             }
-            else if (!string.IsNullOrWhiteSpace(TracesPipeName))
+
+            if (_fileExists(DefaultTracesUnixDomainSocket))
             {
-                traceTransport = TracesTransportType.WindowsNamedPipe;
-
-                TracesPipeTimeoutMs = source?.GetInt32(ConfigurationKeys.TracesPipeTimeoutMs)
-#if DEBUG
-                    ?? 20_000;
-#else
-                    ?? 500;
-#endif
+                TrySetAgentUriAndTransport(UnixDomainSocketPrefix + DefaultTracesUnixDomainSocket);
+                return;
             }
 
-            if (traceTransport == null)
+            ValidationWarnings.Add("No transport configuration found, using default values");
+            TrySetAgentUriAndTransport(DefaultAgentHost, DefaultAgentPort);
+        }
+
+        private bool TrySetAgentUriAndTransport(string host, int port)
+        {
+            return TrySetAgentUriAndTransport($"http://{host}:{port}");
+        }
+
+        private bool TrySetAgentUriAndTransport(string url)
+        {
+            if (!Uri.TryCreate(url, UriKind.RelativeOrAbsolute, out var uri))
```

Comment:
Doesn't this _have_ to be `UriKind.Absolute`? What would a relative `Uri` mean here 🤔